cms/static/js/views/modals/edit_xblock.js

```diff
@@ -119,7 +119,11 @@ define(['jquery', 'underscore', 'gettext', 'js/views/modals/base_modal', 'common
             getTitle: function() {
                 var displayName = this.xblockInfo.get('display_name');
                 if (!displayName) {
-                    displayName = gettext('Component');
+                    if (this.xblockInfo.isSequential() || this.xblockInfo.isChapter() || this.xblockInfo.isVertical()) {
+                        displayName = gettext('Unit');
```

Comment:
I believe it should only say "Unit" for isVertical. The others are Sections and Subsections.

For some odd reason, I can't find the implementation of these methods though.